test/tests/kernels/kernel_precompute/tests

```diff
@@ -1,8 +1,20 @@
 [Tests]
+  issues = 'r2275 #3492 #12658'
+  design = '/kernels/Kernel.md'
+
   [./test]
     type = 'Exodiff'
     input = 'kernel_precompute_test.i'
     exodiff = 'out.e'
     scale_refine = 5
+    requirement = 'MOOSE shall provide optimized kernels for residuals with the test function or its gradient factored out.'
+  [../]
+  [./adtest]
+    type = 'Exodiff'
+    input = 'kernel_precompute_test.i'
```

Comment:
Should be `adkernel_precomute_test.i`? Also I do like to see `PetscJacobianTesters` for our AD stuff